soci/index/gzip_index.go

```diff
@@ -14,10 +14,10 @@
    limitations under the License.
 */
 
-package soci
+package index
```

Comment:
I'm not so sure about this structure. I think what we really want is a full ztoc package.

Reply:
Package soci contains soci index and ztoc definition, building and accessing - this is a full ztoc package. I moved `gzip_index.c` under `index` package to better group it with the C code. The ztoc building logic relies on `gzip_index`. 

Another option would be to simply keep C code under soci, but that bothers a little bit, since we place C files along with Go files, which don't even use the C implementations directly.

Reply:
Index here is confusing. You rename it `sociindex` when importing, but this isn't about soci indices, it's about gzip indices within a zTOC. I think we had discussed calling this `zindex` at one point which is a little more clear. This change also pushes `FileSize` down into `index`, but even gzip zindices don't *really* care about file sizes, that just happens to be the type we're using to represent offsets. I think it would be more clear to create a `ztoc` package that contains all the zTOC code, including the gzip code. 

Overall, I think we should be moving towards separating zTOC from SOCI. zTOCs are the technology that enables SOCI, but they make sense as a concept on their own.